### ISSUE TITLE
Support non-array input query?

### ISSUE THREAD

**Issue by christianbundy** [dominictarr/map-filter-reduce#6]
Hi!

I think this module currently requires array input, like:

```javascript
[{
  "$filter": {
    "foo": "bar"
  }
}]
```

Would you be open to accepting a non-array input? My context is that I'm working on a muxrcpli-like module and I'd like to do:

```shell
ssb query read --query.\$filter.value.channel solarpunk
```

This *could* be resolved by https://github.com/ssbc/ssb-query/issues/15, but I've been bitten by the "you must pass an array but it doesn't error if you pass an object" thing many times before.

**Comment by dominictarr** [comment]
`sbot query.read --query[0]\$filter.value.channel solarpunk` works

**Comment by dominictarr** [comment]
I'm not against making arrays optional though.

**Comment by dominictarr** [comment]
how are you writing these commands?
`ssb query read` it should be `ssb-server query.read`

**Comment by christianbundy** [comment]
I don't think the `[0]` trick works for my command-line parser, and since the query type is a string (https://github.com/ssbc/ssb-query/issues/15) that's what gets passed regardless of what you give as input. I'll put together a PR to handle plain objects too. :+1:

**Comment by dominictarr** [comment]
> since the query type is a string

do you mean this is being enforced somewhere?

**Comment by christianbundy** [comment]
Yeah, Yargs is very particular about the type you pass, so if you declare it as an array and pass "foo" then I think it'll call the method with `["foo"]`. In a similar vein, if you declare that an option is required (like [this one](https://github.com/ssbc/ssb-db/pull/273#discussion_r349837520)) then it won't let you call the function without it.
